lib/assets/javascripts/turbograft/page.coffee

```diff
@@ -25,3 +25,42 @@ Page.refresh = (options = {}, callback) ->
 
 Page.open = ->
   window.open(arguments...)
+
+# Providing hooks for objects to set up destructors:
+onReplaceCallbacks = []
+
+# e.g., Page.onReplace(node, unbindListenersFnc)
+# unbindListenersFnc will be called if the node in question is partially replaced
+# or if a full replace occurs.  It will be called only once
+Page.onReplace = (node, callback) ->
+  throw new Error("Node and callback must both be specified") if !node || !callback
+  onReplaceCallbacks.push({node, callback})
+
+# from http://davidwalsh.name/check-parent-node
+contains = (parentNode, childNode) ->
+  if parentNode.contains
+    parentNode.contains(childNode)
+  else
+    parentNode.compareDocumentPosition(childNode) % 16
+
+document.addEventListener 'page:before-partial-replace', (event) ->
```

Comment:
this is the block that wasn't straightforward to convert, as we don't have lodash nor jquery here

But I think it may be slightly more performant now with the break